application/src/main/kotlin/application/VirtualServiceCommand.kt

```diff
@@ -103,21 +114,35 @@ class VirtualServiceCommand  : Callable<Int> {
         logger.log("${newLine}Injecting the data read from the following stub files into the stub server's state..".prependIndent("  "))
         this.forEach { logger.log(it.filePath.orEmpty().prependIndent("  ")) }
     }
-}
-
-
-
-
-
-
-
-
-
-
-
-
-
-
 
+    companion object {
+        fun virtualServiceValidationRuleset(scenarios: List<Scenario>): MutableList<String> {
+            val errors: MutableList<String> = mutableListOf()
+            val supportedMethods = setOf("GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS")
+            scenarios.forEach{ s ->
+                // Check if the scenario is a valid HTTP method
+                if(s.httpRequestPattern.method !in supportedMethods){
```

Comment:
This line of code (`s.httpRequestPattern.method`) breaks the "law of demeter". Read up on this and fix the issue. Same for other places a similar dot-dot-dot pattern is found.